image_segmentation/pytorch/data_loading/data_loader/unet3d_data_loader.py

```diff
@@ -0,0 +1,139 @@
+            lbls_train.append(case_lbl)
+    mllog_event(key="train_samples", value=len(imgs_train), sync=False)
+    mllog_event(key="eval_samples", value=len(imgs_val), sync=False)
+    imgs_val, lbls_val = split_eval_data(imgs_val, lbls_val, num_shards, shard_id)
+    return imgs_train, imgs_val, lbls_train, lbls_val
+
+
+class SyntheticDataset(Dataset):
+    def __init__(
+        self,
+        channels_in=1,
+        channels_out=3,
+        shape=(128, 128, 128),
+        device="cpu",
+        layout="NCDHW",
+        scalar=False,
+    ):
+        shape = tuple(shape)
+        x_shape = (channels_in,) + shape if layout == "NCDHW" else shape + (channels_in,)
+        self.x = torch.rand(
+            (32, *x_shape), dtype=torch.float32, device=device, requires_grad=False
+        )
+        if scalar:
+            self.y = torch.randint(
+                low=0,
+                high=channels_out - 1,
+                size=(32, *shape),
+                dtype=torch.int32,
+                device=device,
+                requires_grad=False,
+            )
+            self.y = torch.unsqueeze(self.y, dim=1 if layout == "NCDHW" else -1)
+        else:
+            y_shape = (channels_out,) + shape if layout == "NCDHW" else shape + (channels_out,)
+            self.y = torch.rand(
+                (32, *y_shape), dtype=torch.float32, device=device, requires_grad=False
+            )
+
+    def __len__(self):
+        return 64
+
+    def __getitem__(self, idx):
+        return self.x[idx % 32], self.y[idx % 32]
+
+
+def get_data_loaders(flags: Namespace, num_shards: int, global_rank: int, device: torch.device):
+    """
+    Initializes and returns (train_data_loader, val_data_loader)
+
+    :param Namespace flags: the runtime arguments
+    :param int num_shards: number of shards for the train dataset
+    :param int global_rank: global rank associated with the device
+    :param torch.device device: the device to use for MpDeviceLoader
+    :return: the tuple (train_loader, val_loader)
+    :rtype: Union[Tuple[pl.MpDeviceLoader, pl.MpDeviceLoader], Tuple[DataLoader, DataLoader]]
+    """
+    if flags.loader == "synthetic":
+        train_dataset = SyntheticDataset(scalar=True, shape=flags.input_shape, layout=flags.layout)
+        val_dataset = SyntheticDataset(
+            scalar=True, shape=flags.val_input_shape, layout=flags.layout
+        )
+
+    elif flags.loader == "pytorch":
+        x_train, x_val, y_train, y_val = get_data_split(
+            flags.data_dir, num_shards, shard_id=global_rank
+        )
+        train_data_kwargs = {
+            "patch_size": flags.input_shape,
+            "oversampling": flags.oversampling,
+            "seed": flags.seed,
+        }
+        train_dataset = PytTrain(x_train, y_train, **train_data_kwargs)
+        val_dataset = PytVal(x_val, y_val)
+    else:
+        raise ValueError(f"Loader {flags.loader} unknown. Valid loaders are: synthetic, pytorch")
+
+    if flags.torch_xla:
```

Comment:
did you consider `flags.device == "xla"`?

Reply:
To confirm, do you mean getting rid of the `torch_xla` argparse argument and replace it with a new `str` argparse argument called `device`? Currently, my argparse parser does not have a `device` argument.

I am not sure how `flags.device` would be better than `flags.torch_xla`. Could you please clarify what you meant?

Reply:
`torch_xla` is a binary choice and not flexible/extensible to other backends in the future. a device flag is more flexible.

Reply:
I see. A device flag is indeed more flexible. I have replaced all occurrences of `flags.torch_xla` with `flags.device` now.